lib/SPIRV/SPIRVUtil.cpp

```diff
@@ -1863,6 +1863,97 @@ bool lowerBuiltinVariablesToCalls(Module *M) {
   return true;
 }
 
+bool postProcessBuiltinReturningStruct(Function *F) {
+  Module *M = F->getParent();
+  LLVMContext *Context = &M->getContext();
+  std::string Name = F->getName().str();
+  F->setName(Name + ".old");
+  for (auto *U : F->users()) {
+    if (auto *CI = dyn_cast<CallInst>(U)) {
+      auto *ST = dyn_cast<StoreInst>(*(CI->user_begin()));
+      assert(ST);
```

Comment:
NIT: `cast<>` is equivalent to `dyn_cast<> + assert()`.
```suggestion
      auto *ST = cast<StoreInst>(*(CI->user_begin()));
```

Reply:
Done, thanks